### PR TITLE
fix(A11Y-55): Screen Reader: Core Player: CTA Overlay dialog is not announced by screen reader

### PR DESCRIPTION
**Pull request by SivanA-Kaltura**
### Description of the Changes

- Add role="alertdialog" to overlay so it would be announced when it appears
- Add focus handling on buttons (focus the first visible button)
- Add focus trap to the overlay

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/call-to-action-overlay/call-to-action-overlay.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const CallToActionOverlay = withText({closeLabel: 'overlay.close'})(
   connect(mapStateToProps)(
     ({title, description, buttons, onClose, onClick, closeLabel, descriptionLines, sizeClass}: CallToActionOverlayProps): any => {
       return (
-        <div data-testid="call-to-action-overlay" className={`${styles.callToActionOverlay} ${sizeClass}`}>
+        <div role="alert" data-testid="call-to-action-overlay" className={`${styles.callToActionOverlay} ${sizeClass}`}>
```
Comment (semarche-kaltura) [inline src/components/call-to-action-overlay/call-to-action-overlay.tsx:60]: As I understand `<CallToActionOverlay />` renders dynamically, does screen-readers handles changes for such nodes properly?

Comment (SivanA-Kaltura) [reply src/components/call-to-action-overlay/call-to-action-overlay.tsx:60]: the screen reader notifies when the overlay appears and reads out the title, and when the overlay shows again with a different title it reads the new title. this covers the issue, as i understandv it

File: src/components/call-to-action-overlay/call-to-action-overlay.tsx [hunk 2]
```diff
@@ -57,30 +58,34 @@ const CallToActionOverlay = withText({closeLabel: 'overlay.close'})(
...
-              tooltip={{label: closeLabel!}}
-              ariaLabel={closeLabel}
-              icon={'close'}
-            />
-          </div>
-          <div className={styles.content}>
-            <div className={styles.title}>
-              <TextWithTooltip text={title || ''} numberOfLines={1} />
+        <div role="alertdialog" data-testid="call-to-action-overlay" className={`${styles.callToActionOverlay} ${sizeClass}`}>
+          <FocusTrap active>
```
Comment (semarche-kaltura) [inline src/components/call-to-action-overlay/call-to-action-overlay.tsx:62]: I've found that playkit-js-ui already has focus-trap utility, could we re-use it in CTA as well?
https://github.com/kaltura/playkit-js-ui/blob/master/src/utils/popup-keyboard-accessibility.tsx#L9

usages
https://github.com/kaltura/playkit-js-info/blob/master/src/components/info/index.tsx#L47
https://github.com/kaltura/playkit-js-moderation/blob/master/src/components/moderation/moderation.tsx#L84

**Review by semarche-kaltura — CHANGES_REQUESTED** [review]

File: src/components/call-to-action-overlay/call-to-action-overlay.tsx (hunk 2 above)
Comment (SivanA-Kaltura) [reply src/components/call-to-action-overlay/call-to-action-overlay.tsx:62]: interesting, will check it out. thanks.

**Review by semarche-kaltura — APPROVED** [review]
discussing with @SivanA-Kaltura about advantages and disadvantages of use current solution or `withKeyboardA11y` utility from player-ui
We agreed that current solution easier and we would like test it on the CTA plugin. If current solution works fine - later we'll implement the solution with `<FocusTrap>` hoc to other plugins.
The `<Overlay />` component from palyer-ui needs to be refactored to handle issues with styles and close on `Esc` button

**Pull request merged**